### PR TITLE
Reject ID collisions in identity map unconditionally

### PR DESCRIPTION
**Pull request by greg0ire**

### CODE REVIEW

**Comment by greg0ire** [comment]
Cc @mpdude :slightly_smiling_face:

File: lib/Doctrine/ORM/Configuration.php [hunk 1]
```diff
@@ -605,13 +605,18 @@ public function setLazyGhostObjectEnabled(bool $flag): void
         $this->attributes['isLazyGhostObjectEnabled'] = $flag;
     }
 
+    /** To be deprecated in 3.1.0 */
     public function setRejectIdCollisionInIdentityMap(bool $flag): void
```
Comment (derrabus) [inline lib/Doctrine/ORM/Configuration.php:609]: We should throw if `$flag === false`.

**Review by derrabus — CHANGES_REQUESTED** [review]

File: lib/Doctrine/ORM/Configuration.php (hunk 1 above)
Comment (greg0ire) [reply lib/Doctrine/ORM/Configuration.php:609]: done

**Review by derrabus — APPROVED** [review]

**Comment by mpdude** [comment]
Thank you :+1:

**Pull request merged**